### PR TITLE
feat: Adding discussion to proposal flow and pages

### PR DESCRIPTION
**Pull request by rossneilson**
# Description

The new flow for governance is: Create discussion -> Create proposal

Discussions can have multiple proposals created from them

Our proposals now reference an orbis/ceramic stream ID so we can easily link discussions and proposals and do more complex queries

RIght now its very simple and we will later add reactions, links and more

Closes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [ ] My code follows the existing style of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by Filipv95** [comment]
Hey @rossneilson I've checked this PR, found this:

1. Created `Discussion` -> Go to `Proposal Creation` page -> Try adding any action -> **Getting white page**
2. We have this edge case where I can still open `Proposal Creation` page without creating Discussion first:
If I go to `All Proposals` page from any Guild -> `Create Proposal` button is visible -> Go to `Proposal Creation` page -> Try adding any action -> **Getting white page**
3. Was able to create Proposal without adding any action

white-page
proposal-no-action

**Comment by Filipv95** [comment]
Just to check, I was not able to see what `Proposal` was created from particular `Discussion` (on that Discussion page) is this expected behavior?

**Comment by rossneilson** [comment]
> Hey @rossneilson I've checked this PR, found this:
> 
> 
> 
> 1. Created `Discussion` -> Go to `Proposal Creation` page -> Try adding any action -> **Getting white page**
> 
> 2. We have this edge case where I can still open `Proposal Creation` page without creating Discussion first:
> 
> If I go to `All Proposals` page from any Guild -> `Create Proposal` button is visible -> Go to `Proposal Creation` page -> Try adding any action -> **Getting white page**
> 
> 3. Was able to create Proposal without adding any action
> 
> 
> 
> white-page
> 
> proposal-no-action
> 
> 

Pretty sure this was introduced by another PR, @rulfo71 is working on it

**Comment by rulfo71** [comment]
Yes i am! 
@Filipv95 which kind of action were you adding?

**Comment by rossneilson** [comment]
> Just to check, I was not able to see what `Proposal` was created from particular `Discussion` (on that Discussion page) is this expected behavior?

Yep expected behaviour for now, we have the link on the backend and will add it later

**Comment by Filipv95** [comment]
@rossneilson ack, ty!

Hey @rulfo71 I've tried adding every action on localhost, always got `White page` after I click `Save Action` button.

**Comment by rulfo71** [comment]
hey! i'm seeing the picture now and this is the simulation error @rossneilson. this is fixed by just adding a "?". just in case you want to send a version fixing only this. 

the fix i am doing is for Vesting Contracts

**Comment by Filipv95** [comment]
@rulfo71 ack, ty! All good then.

File: src/components/DiscussionCard/DiscussionCard.tsx [hunk 1]
```diff
@@ -2,31 +2,37 @@ import { Flex } from 'components/primitives/Layout';
 import { UnstyledLink } from 'components/primitives/Links';
 import { shortenAddress } from 'utils';
 import { CardTitle, MainWrapper, FooterElement } from './DiscussionCard.styled';
-import { BiHeart } from 'react-icons/bi';
-import { MdReply } from 'react-icons/md';
+// import { BiHeart } from 'react-icons/bi';
+// import { MdReply } from 'react-icons/md';
```
Comment (rulfo71) [inline src/components/DiscussionCard/DiscussionCard.tsx:6]: can we remove this comments?

File: src/components/DiscussionCard/DiscussionCard.tsx [hunk 2]
```diff
@@ -2,31 +2,37 @@ import { Flex } from 'components/primitives/Layout';
...
             {ensName || shortenAddress(creatorAddress)}
           </FooterElement>
-          <FooterElement>
+          {/* <FooterElement>
             <BiHeart size="20px" /> {discussion?.count_likes}
           </FooterElement>
           <FooterElement>
             <MdReply size="20px" /> {discussion?.count_replies}
-          </FooterElement>
+          </FooterElement> */}
```
Comment (rulfo71) [inline src/components/DiscussionCard/DiscussionCard.tsx:35]: also here

**Review by rulfo71 — APPROVED** [review]
looks good! 
i tried the discussions and works great. i left some minor comments

File: src/components/DiscussionCard/DiscussionCard.tsx (hunk 1 above)
Comment (rossneilson) [reply src/components/DiscussionCard/DiscussionCard.tsx]: Done

File: src/components/Forum/types.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export interface DiscussionContent {
   master?: string | null;
   replyTo?: string | null;
   mentions?: string[];
-  data?: Record<string, string>;
+  data?: any;
```
Comment (MiltonTulli) [inline src/components/Forum/types.ts:17]: Can we get types for this or is too hard?

File: src/components/GuildSidebar/GuildSidebar.tsx [hunk 4]
```diff
@@ -73,7 +74,12 @@ export const GuildSidebar: React.FC<GuildSidebarProps> = ({
             {t('allProposals')}
           </SidebarMenuItem>
         </UnstyledLink>
-        <SidebarMenuItem>{t('settings')}</SidebarMenuItem>
+        <UnstyledLink to={locations.allDiscussions}>
+          <SidebarMenuItem current={pathname === locations.allDiscussions}>
+            {t('allDiscussions')}
+          </SidebarMenuItem>
+        </UnstyledLink>
+        {/* <SidebarMenuItem>{t('settings')}</SidebarMenuItem> */}
```
Comment (MiltonTulli) [inline src/components/GuildSidebar/GuildSidebar.tsx:82]: Remove here

File: src/hooks/Guilds/useProposalMetadata.ts [hunk 5]
```diff
@@ -1,27 +1,59 @@
...
-  }, [proposal]);
+  }, [proposal?.contentHash]);
+
+  // Get orbis data
+  useEffect(() => {
+    let data;
+    if (proposal?.contentHash?.startsWith('streamId://')) {
+      const fetchData = async () => {
+        data = await orbis.getPost(proposal?.contentHash.slice(11));
+        console.log({ data });
```
Comment (MiltonTulli) [inline src/hooks/Guilds/useProposalMetadata.ts:31]: We can remove this log

**Review by MiltonTulli — DISMISSED** [review]
Looks good to me. Just minor comments

File: src/components/Forum/types.ts (hunk 3 above)
Comment (rossneilson) [reply src/components/Forum/types.ts:17]: The type on orbis is actually just anything but maybe we can write our own in future. Will leave it as any for now though

**Comment by dcrescimbeni** [comment]
This looks OK.

There's a bug that is related to the previous implementation where, after switching wallets, it reconnects with the wallet you first connect with.

To reproduce: change wallets and add comments, they all get posted with the same address.

Should we fix it in this PR or make an issue to solve it later?

**Comment by rossneilson** [comment]
> This looks OK.
> 
> There's a bug that is related to the previous implementation where, after switching wallets, it reconnects with the wallet you first connect with.
> 
> To reproduce: change wallets and add comments, they all get posted with the same address.
> 
> Should we fix it in this PR or make an issue to solve it later?

Nice, lets make it another issue. If anyone can fix it before release then we can include it in the release

**Comment by dcrescimbeni** [comment]
Looks good! Everything working as expected

**Review by dcrescimbeni — APPROVED** [review]

**Pull request merged**